src/GameData.ts

```diff
@@ -0,0 +1,240 @@
+    name: string;
+    categories: string[];
+    price: number;
+    priceTotal: number;
+    iconPath: number;
+    from: string[];
+    to: string[];
+}
+
+export default class GameData {
+
+    private champData: ChampionData[];
+    private perkData: PerkData[];
+    private itemData: ItemData[];
+
+    private bot: Discord.Client;
+
+    public constructor(bot: Discord.Client) {
+        this.bot = bot;
+
+        bot.on("ready", async () => {
+            setInterval(this.reloadData, 86400 * 1000);
+        });
+    }
+
+    public async reloadData() {
+        this.champData = await this.loadChampionData();
+        this.perkData = await this.loadPerkData();
+        this.itemData = await this.loadItemData();
+        console.log("Game data reloaded!");
+    }
+
+    public async loadChampionData(): Promise<ChampionData[]> {
+        const returnData: ChampionData[] = [];
+
+        const championDataUrl = "https://raw.communitydragon.org/pbe/plugins/rcp-be-lol-game-data/global/default/v1/champion-summary.json";
+        const champions = await (await fetch(championDataUrl)).json();
+        champions.forEach((c: any) => returnData.push({
+            id: c.id,
+            name: c.name,
+            key: c.alias,
+            skins: [],
+        }));
+
+        const skinDataUrl = "https://raw.communitydragon.org/pbe/plugins/rcp-be-lol-game-data/global/default/v1/skins.json";
+        const skins = await (await fetch(skinDataUrl)).json() as SkinData[];
+
+        (champions as ChampionData[]).map(c => c.id)
+            .filter(id => id > 0)
+            .map(i => i.toString())
+            .forEach(id => {
+                const skinKeys = Object.keys(skins).filter(k => k.startsWith(id)).filter(k => (k.length - id.length) === 3).map(k => +k);
+
+                skinKeys.forEach(key => {
+                    const data = skins[key];
+
+                    const chromaData: ChromaData[] = [];
+                    if (data.hasOwnProperty("chromas")) {
+                        data.chromas.forEach(chroma => {
+                            chromaData.push(chroma);
+                        });
+                    }
+
+                    const skinData: SkinData = {
+                        id: data.id,
+                        name: data.name,
+                        splashPath: data.splashPath,
+                        uncenteredSplashPath: data.uncenteredSplashPath,
+                        tilePath: data.tilePath,
+                        loadscreenPath: data.loadscreenPath,
+                        chromas: chromaData,
+                    };
+
+                    returnData.filter(r => r.id === +id)[0].skins.push(skinData);
+                });
+            });
+
+        return returnData;
+    }
+
+    public async  loadPerkData(): Promise<PerkData[]> {
+        const returnData: PerkData[] = [];
+
+        const perkDataUrl = "https://raw.communitydragon.org/pbe/plugins/rcp-be-lol-game-data/global/default/v1/perks.json";
```

Comment:
Make with an actual const pls. If this url might be used in multiple files, make it a global const or static class, or w.e is the standard in js.

Reply:
We have setting files for this.